Dispatcher/Controllers/Homepage/HomeViewController.swift

```diff
@@ -23,11 +26,43 @@ class HomeViewController: UIViewController {
         articlesTableView.dataSource = self
         articlesTableView.delegate = self
         
-        articlesTableView.register(UINib(nibName: "ArticleCell", bundle: nil), forCellReuseIdentifier: ArticleCell.articleCellIdentifier)
+        articlesTableView.register(UINib(nibName: AppConstants.articleCellNibName, bundle: nil), forCellReuseIdentifier: AppConstants.articleCellIdentifier)
+        configureActivityIndicator()
+        fetchArticles()
+    }
+    
+    private func showErrorAlert(_ msg: String) {
         
+        let alertController = UIAlertController(title: "Error", message: msg, preferredStyle: .alert)
+        let okAction = UIAlertAction(title: "OK", style: .default, handler: nil)
+        
+        alertController.addAction(okAction)
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    private func configureActivityIndicator() {
+        activityIndicator.style = .large
+        activityIndicator.center = homepageView.center
+        homepageView.addSubview(activityIndicator)
+    }
+    
+    private func fetchArticles() {
```

Comment:
Following the CleanCode principles, we should stick to a single phrase for describing the action of retrieving data from the api. Either choose "fetch" or "get", but make sure to use the chosen word throughout the project in all related function names